Analysis/indiv_gals.py

```diff
@@ -1,95 +1,79 @@
-        and it holds individual galaxy line measurements, bin temperatures (for both cases) applied to the
-        galaxies within each bin, detection markings, and individual and bin ID numbers.
-        
-    Usage:
-        indiv_gals.create_Te_line_table(fitspath, line_file, mass_bin_file, HB_bin_file, mass_Te_file, HB_Te_file)
+        This function creates a table containing bin information such as bin ID numbers, minimum, 
+        maximum, median, and average mass and/or HBeta luminosity measurements, bin detection markings,
+        and indiviudal galaxy ID numbers.
         
     Params:
-        fitspath --> a string of the file path where the input file is and where the output file will be placed.
-        line_file --> a fits table containing the individual line measurements.
-        mass_bin_file --> an npz file containing the indices of galaxies, relative to line_file, contained in 
-            each mass bin. It also contains the mass values of all sources (indexed the same as line_file).
-        HB_bin_file --> an npz file containing the indices of galaxies, relative to line_file, contained in 
-            each mass-LHbeta bin. It also contains the log of the HBeta luminosity values of all sources.
-        mass_Te_file --> an ascii table containing the electron temperatures and detection markings for each
-            mass bin.
-        HB_Te_file --> an ascii table containing the electron temperatures and detection markings for each
-            mass-LHbeta bin.
+        fitspath --> file path where bin files are located and where the output file will be located
+        line_file --> a fits table containing individual galaxy ID numbers
+        use_revised (OPTIONAL) --> boolean to use the revised validation table or not (default is False)
         
     Returns:
         None
         
     Outputs:
-        fitspath + 'individual_Te_emLines.tbl' --> an ascii table containing individual and bin ID numbers,
-            bin temperatures correlated to individual galaxies in each bin, detection markings, and individual
-            line fluxes and signal to noise.
+        'individual_bin_info.tbl'
     '''
     
-    if LHb_bin == True:
-        bin_type = 'logLHb'
-    else:
-        bin_type = 'logM'
-    
-    bin_npz = np.load(bin_npz_file)
-    bin_tbl = asc.read(fitspath + filename_dict['bin_info'])
-    valid_tbl = asc.read(fitspath + filename_dict['bin_valid'])
+    bin_npz = np.load(fitspath + filename_dict['bin_info'].replace('.tbl', '.npz'))
+    tbl = asc.read(fitspath + filename_dict['bin_info'])
     hdu = fits.open(line_file)
+    if use_revised == True:
+        valid_tbl = asc.read(fitspath + filename_dict['bin_valid_rev'])
+    else:
+        valid_tbl = asc.read(fitspath + filename_dict['bin_valid'])
+        
+    #Convert bin info table into a dictionary
+    cols = tbl.colnames
+    bin_dict = {}
+    for col in cols:
+        if col not in bin_dict:
```

Comment:
Not sure if there is a case where `cols` has duplicate entries.  This is obtained from astropy table, so unlikely.